### DISCUSSION TITLE
Add build option to call getStaticProps on every request

### DISCUSSION THREAD

**Discussion by InsOpDe** [vercel/next.js#28926] [Ideas]
### Describe the feature you'd like to request

Add a build parameter or a next-config where I can set an option to build next with getStaticProps being called on every request **and not on build time**. Like its already being done when you run `next dev`

Why? Because I cannot run `next build` inside the CI pipeline because I dont want to retrieve data from external sources (because they might change). 

And because I cannot use `next build` I wont get the possible errors reported which I will eventually get when I am deploying the app.

**So what I need is to smoke test the build of my nextjs application**

because otherwise errors like:
> Global CSS cannot be imported from files other than your Custom <App>. Please move all global CSS imports to pages/_app/index.tsx. Or convert the import to Component-Level CSS (CSS Modules).

Getting discovered when I build & deploy

### Describe the solution you'd like

As stated above one of two:
* [ ] `next build --smoke` to build nextjs with all `getStaticProps` exchanged with `getServerSideProps` (like its being done with `next dev`
* [ ] a `next.config.js` configuration where I can disable `getStaticProps` and fallback to `getServerSideProps`

### Describe alternatives you've considered

Current alternative is to remove all `getStaticProps` from the application and replace it with `getServerSideProps` then run `next build` inside the pipeline to check for build errors.

**Comment by mzaien** [comment]
This is what getServerSideProps do
 right?

**Reply by InsOpDe** [reply]
that is correct. You do not expect me to rewrite the code every time I want to do a smoke test (like I described above) ?

**Reply by mzaien** [reply]
excuse me did not read the whole idea, now i've read it is a great idea having the best performance with the flexibility if server render
